### PR TITLE
Jon tidy pytests

### PR DESCRIPTION
**Pull request by jonblower**
This PR does a few things:
- Stops flake8 complaining about some formatting
- Removes some duplicate code
- Converts the `test` and `tools` directories into modules by adding `__init__.py`.
- Sorts out relative imports relating to the above.
- Checks in a working configuration for Visual Studio Code

These changes mean that the pytests are correctly detected and run by Visual Studio Code (previously I was getting all kinds of importing errors, although the tests worked fine on the command line outside of VSCode).

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    "python.linting.flake8Enabled": true,
+    "python.linting.flake8Args": [
+        "--extend-ignore=E201,E202,E203",
+        "--max-line-length=120"
+    ],
+    "python.testing.pytestEnabled": true,
+    "python.testing.pytestArgs": [
+        "--rootdir",
+        "${workspaceFolder}/test"
+    ],
```
Comment (letmaik) [inline .vscode/settings.json:11]:
```suggestion
    "python.testing.pytestArgs": [
        "--rootdir",
        "${workspaceFolder}/test"
    ],
```
```suggestion
```

I tried your variant out locally but VS Code wasn't able to run individual tests with it. Removing it made it work for all tests and individually selected tests.

File: README.md [hunk 2]
```diff
@@ -13,11 +13,11 @@ N.B. Make sure to install requirements via `pip`, not `conda` (at the time of wr
 ## Running the validator
 
 ```sh
-python tools/validator.py coverage.covjson
+python tools/validator.py my.covjson
```
Comment (letmaik) [inline README.md:16]:
```suggestion
python tools/validator.py my.covjson
```
```suggestion
python -m tools.validator my.covjson
```

`validator.py` doesn't use relative imports, but `bundle_schema.py` now does, and the only consistent way to run scripts with relative imports is to use `python -m` and treat them as modules.  The same change has to be made in `.github\workflows\publish-schema.yml` when invoking the `tools/` scripts. That workflow only runs after merging, on `main`.

Comment (letmaik) [inline README.md:16]:
```suggestion
python tools/validator.py my.covjson
```
```suggestion
python -m tools.validator my.covjson
```

`validator.py` doesn't use relative imports, but `bundle_schema.py` now does, and the only consistent way to run scripts with relative imports is to use `python -m` and treat them as modules.  The same change has to be made in `.github\workflows\publish-schema.yml` when invoking the `tools/` scripts. That workflow only runs after merging, on `main`.

**Review by letmaik — CHANGES_REQUESTED** [review]
Looks good, I never used the pytest runner of the VS Code Python extension but can confirm it works. Some minor changes needed, esp. in `.github\workflows\publish-schema.yml`, see comments.

File: README.md (hunk 2 above)
Comment (jonblower) [reply README.md:16]: Strangely, both ways of invoking the validator work for me, but happy to change if it makes it more portable

File: .vscode/settings.json (hunk 1 above)
Comment (jonblower) [reply .vscode/settings.json:11]: Strangely, removing this made it stop working for me! I've added a commit that invokes `pytest` with `python -m`. Does that work for you?

File: .github/workflows/publish-schema.yml [hunk 3]
```diff
@@ -26,10 +26,10 @@ jobs:
       run: mkdir dist
       
     - name: Bundle schema
-      run: python tools/bundle_schema.py --out dist/schema.json
+      run: python python -m tools.bundle_schema --out dist/schema.json
```
Comment (letmaik) [inline .github/workflows/publish-schema.yml:29]:
```suggestion
      run: python -m tools.bundle_schema --out dist/schema.json
```

File: .github/workflows/publish-schema.yml [hunk 4]
```diff
@@ -26,10 +26,10 @@ jobs:
...
     
     - name: Downgrade schema to draft-07
-      run: python tools/downgrade_schema_to_draft07.py dist/schema.json
+      run: python python -m tools.downgrade_schema_to_draft07 dist/schema.json
```
Comment (letmaik) [inline .github/workflows/publish-schema.yml:32]:
```suggestion
      run: python -m tools.downgrade_schema_to_draft07 dist/schema.json
```

File: .vscode/settings.json (hunk 1 above)
Comment (letmaik) [reply .vscode/settings.json:11]: Unfortunately not, the "Python test log" output says:
```
============================= test session starts =============================
platform win32 -- Python 3.9.5, pytest-7.1.1, pluggy-1.0.0
rootdir: c:\Users\maik\git\covjson-validator\test
collected 0 items

- generated xml file: C:\Users\maik\AppData\Local\Temp\tmp-23440k5B13YprZ3lR.xml -
============================ no tests ran in 0.03s ============================
ERROR: file or directory not found: ./domain_types/test_grid.py::test_extra_multi_coordinate_axis

ERROR conda.cli.main_run:execute(33): Subprocess for 'conda run ['python', 'c:\\Users\\maik\\.vscode\\extensions\\ms-python.python-2022.4.0\\pythonFiles\\get_output_via_markers.py', '-m', 'pytest', '--override-ini', 'junit_family=xunit1', '--junit-xml=C:\\Users\\maik\\AppData\\Local\\Temp\\tmp-23440k5B13YprZ3lR.xml', '--rootdir', 'c:\\Users\\maik\\git\\covjson-validator/test', './domain_types/test_grid.py::test_extra_multi_coordinate_axis']' command failed.  (See above for error)
```

When I remove the rootdir option and the pytest path option it looks like this:
```
============================= test session starts =============================
platform win32 -- Python 3.9.5, pytest-7.1.1, pluggy-1.0.0
rootdir: c:\Users\maik\git\covjson-validator
collected 2 items

test\test_axis.py ..                                                     [100%]

- generated xml file: C:\Users\maik\AppData\Local\Temp\tmp-23440zeOm4F7YeTd4.xml -
============================== 2 passed in 0.11s ==============================
```

Comment (jonblower) [reply .vscode/settings.json:11]: OK, I can't find a way to make this consistently work for both of us. I suggest that I just remove `settings.json` from the PR and we'll just keep our locally-working versions.

**Comment by jonblower** [comment]
I think this is ready for re-review now. I've removed settings.json from version control as I can't see a way of making it work consistently for both of us

**Review by letmaik — APPROVED** [review]

**Pull request merged**